### PR TITLE
sign in

### PR DESCRIPTION
**Pull request by HasNahapetyan**

### CODE REVIEW

File: src/main/java/com/example/kidz_school/entity/Role.java [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
 public enum Role {
-    USER,
-    ADMIN
+    USER(Set.of()),
+    ADMIN(Set.of());
```
Comment (hovgevorgyan1994) [inline src/main/java/com/example/kidz_school/entity/Role.java:11]: Why your roles don't have authorities?

File: src/main/java/com/example/kidz_school/entity/Authority.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.example.kidz_school.entity;
+
+public enum Authority {
+}
```
Comment (hovgevorgyan1994) [inline src/main/java/com/example/kidz_school/entity/Authority.java:4]: Why don't you have any authority?

File: src/main/java/com/example/kidz_school/security/JwtAuthenticationEntryPoint.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Component
+public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint, Serializable {
+
+    @Override
+    public void commence(HttpServletRequest request,
+                         HttpServletResponse response,
+                         AuthenticationException authException) throws IOException {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+    }
+}
```
Comment (hovgevorgyan1994) [inline src/main/java/com/example/kidz_school/security/JwtAuthenticationEntryPoint.java:21]: Well you can't just sendError, you need to handle it properly. How does the frontend know about the error and how it shows the error?

File: src/main/java/com/example/kidz_school/controller/MainController.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        if (userByEmail.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        } else if (!passwordEncoder.matches((authRequestDto.getPassword()), userByEmail.get().getPassword())) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        } else if (!userByEmail.get().isEnabled()) {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        } else {
+            String token = jwtService.generateToken(userByEmail.get());
+            session.setAttribute("AuthorizationToken", "Bearer " + token);
+            session.setMaxInactiveInterval(900); //15 minutes
```
[the remark this answers is not in the document]
Comment (HasNahapetyan) [reply src/main/java/com/example/kidz_school/controller/MainController.java:89]: Specifies the time, in seconds

[the remark this answers is not in the document]
Comment (hovgevorgyan1994) [reply src/main/java/com/example/kidz_school/controller/MainController.java:89]: What time and why do you need it?

File: pom.xml [hunk 5]
```diff
@@ -55,12 +57,54 @@
...
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-lambda-java-core</artifactId>
+            <version>${aws-lambda-java-core.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-lambda-java-events</artifactId>
+            <version>${aws-lambda-java-events.version}</version>
+        </dependency>
```
Comment (hovgevorgyan1994) [inline pom.xml:106]: You don't need those dependencies

**Review by hovgevorgyan1994 — APPROVED** [review]

**Pull request merged**